catalog/nsili/catalog-nsili-endpoint/src/main/java/com/connexta/alliance/nsili/endpoint/managers/ProductMgrImpl.java

```diff
@@ -81,17 +85,19 @@ public GetParametersRequest get_parameters(Product prod, String[] desired_parame
     public GetRelatedFilesRequest get_related_files(Product[] products, FileLocation location,
             String type, NameValue[] properties)
             throws ProcessingFault, InvalidInputParameter, SystemFault {
+        String id = UUID.randomUUID().toString();
+
         GetRelatedFilesRequestImpl getRelatedFilesRequest = new GetRelatedFilesRequestImpl();
 
         try {
-            _poa().activate_object_with_id("get_related_files".getBytes(Charset.forName(ENCODING)),
+            _poa().activate_object_with_id(id.getBytes(Charset.forName(ENCODING)),
                     getRelatedFilesRequest);
         } catch (ServantAlreadyActive | ObjectAlreadyActive | WrongPolicy e) {
             LOGGER.warn(
```

Comment:
We can just throw e here, instead of getting the localized message. Also recommend changing this to error level